modules/core/src/main/java/org/apache/ignite/internal/util/Cursor.java

```diff
@@ -25,5 +26,40 @@
  * @param <T> Type of elements.
  */
 public interface Cursor<T> extends Iterator<T>, Iterable<T>, AutoCloseable {
+    /** {@inheritDoc} */
+    @Override
+    default Iterator<T> iterator() {
+        return this;
+    }
 
+    /**
+     * Creates an iterator based cursor.
+     *
+     * @param it Iterator.
+     * @param <T> Type of elements in iterator.
+     * @return Cursor.
+     */
+    static <T> Cursor<T> fromIterator(Iterator<? extends T> it) {
+        return new Cursor<T>() {
+            /** {@inheritDoc} */
+            @Override
+            public void close() throws Exception {
+                if (it instanceof Closeable) {
```

Comment:
I believe `AutoCloseable` fits better here. The logic is simple: since `Cursor` extends `AutoCloseable`, I would expect that wrapped iterator which extends `AutoCloseable` as well will be closed by invocation of 'close()' on the cursor

Reply:
Closeable is a high-level interface, why would we restrict ourselves with just one of its implementations? I don't see a connection

Reply:
Sorry, I'm just stupid and thought that AutoCloseable extends Closeable, which is wrong